addons/mail/static/src/discuss/core/web/discuss_core_web_service.js

```diff
@@ -43,6 +43,17 @@ export class DiscussCoreWeb {
                 );
             }
         });
+        this.busService.subscribe("discuss.channel/mentioned", async (payload) => {
+            const { channel, mentioned_by_user_id: mentionedByUserId } = payload;
+            const thread = this.store.Thread.insert(channel);
+            await thread.fetchChannelInfo();
+            if (mentionedByUserId && mentionedByUserId !== this.store.self.userId) {
+                this.notificationService.add(
+                    _t("You have been mentioned to #%s", thread.displayName),
+                    { type: "info" }
+                );
+            }
+        });
```

Comment:
- should notify to out_of_focus service so it follows the push notification settings. Yes, the service is poorly named 💀
- Instead of `fetchChannelInfo`, pass the channel name in the `mail.record/insert` notification